api/src/main/java/org/pmiops/workbench/billing/InitialCreditsService.java

```diff
@@ -242,6 +258,155 @@ public double getWorkspaceCreatorFreeCreditsRemaining(DbWorkspace dbWorkspace) {
     return Math.max(creatorFreeCreditsRemaining, 0);
   }
 
+  public void checkCreditsExpirationForUserIDs(List<Long> userIdsList) {
+    if (userIdsList != null && !userIdsList.isEmpty()) {
+      Iterable<DbUser> users = userDao.findAllById(userIdsList);
+      users.forEach(this::checkExpiration);
+    }
+  }
+
+  public Optional<Timestamp> getCreditsExpiration(DbUser user) {
```

Comment:
is this need to be public or just an internal private method

Reply:
This is used in CommonMappers, so I would prefer to keep it public. If you have a different idea though, I am open to it though.